### PR TITLE
Rule/lcom

### PR DESCRIPTION
**Pull request by Mkohm**

### CODE REVIEW

File: src/main/kotlin/io/github/mkohm/detekt/hint/demo/LackOfCohesionMethodsRule.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package io.github.mkohm.detekt.hint.demo
+
+class Foo {
```
Comment (Detekt-hintBot) [inline src/main/kotlin/io/github/mkohm/detekt/hint/demo/LackOfCohesionMethodsRule.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Mkohm/detekt-hint/blob/7e931974fdcda7e78d88743129a7c1a58086cadc/src/main/kotlin/io/github/mkohm/detekt/hint/demo/LackOfCohesionMethodsRule.kt#L3"></span>The file name ‘LackOfCohesionMethodsRule.kt’ does not match the name of the single top-level declaration ‘Foo’.</td>
    </tr>
  </tbody>
</table>

File: src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    )
+    private val thresholdValue =
+        valueOrNull<String>("threshold")?.toDouble() ?: error("You must specify a threshold value in detekt.yml")
+
+    /**
+     * The number of properties in the class. Includes both properties that comes
+     */
+    private var propertyCount: Int = 0
+    private lateinit var publicMethods: List<KtNamedFunction>
+    // todo: consider starting at 1
```
Comment (Detekt-hintBot) [inline src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Mkohm/detekt-hint/blob/7e931974fdcda7e78d88743129a7c1a58086cadc/src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt#L38"></span>Private variable names should match the pattern: (_)?[a-z][A-Za-z0-9]*</td>
    </tr>
  </tbody>
</table>

Comment (Detekt-hintBot) [inline src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Mkohm/detekt-hint/blob/7e931974fdcda7e78d88743129a7c1a58086cadc/src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt#L38"></span>This comment contains text that has been defined as forbidden in detekt.</td>
    </tr>
  </tbody>
</table>

File: src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        property: KtNamedDeclaration,
+        publicMethod: KtNamedFunction,
+        foundCallees: List<KtNamedFunction>
+    ): ArrayList<KtNamedFunction> {
+
+        // If we have any call expressions inside the parent function with the same name as the parent function - There are some possibilities;
+        // 1. calling another function with the same name, but with different signature
+        // 2. calling the supermethod
+        // 3. calles itself - we have recursion.
+        // Todo: create good test cases that will cover these different edge-cases
```
Comment (Detekt-hintBot) [inline src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Mkohm/detekt-hint/blob/992c6e426a96c8391cea6e3713e22ff0510b123f/src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt#L144"></span>This comment contains text that has been defined as forbidden in detekt.</td>
    </tr>
  </tbody>
</table>

File: src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    // todo: consider starting at 1
+    private var mf_sum = 0
+
+    override fun visitNamedDeclaration(declaration: KtNamedDeclaration) {
+        super.visitNamedDeclaration(declaration)
+
+        if ((declaration is KtProperty || isPropertyFromPrimaryConstructor(declaration)) && propertyIsDeclaredInsideClass(declaration)) {
+            propertyCount++
+
+            val containingClass = declaration.containingClass()!!
```
Comment (Detekt-hintBot) [inline src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Mkohm/detekt-hint/blob/7e931974fdcda7e78d88743129a7c1a58086cadc/src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt#L47"></span>Calling !! on a nullable type will throw a NullPointerException at runtime in case the value is null. It should be avoided.</td>
    </tr>
  </tbody>
</table>

File: src/main/kotlin/io/github/mkohm/detekt/hint/demo/Rectangle.kt [hunk 5]
```diff
@@ -15,5 +15,11 @@ open class Rectangle(width: Int, height: Int) {
...
     fun setHeight(newHeight: Int) {
         this.height = newHeight
+
+        lol()
+    }
+
+    private fun lol() {
+
     }
 }
```
Comment (Detekt-hintBot) [inline src/main/kotlin/io/github/mkohm/detekt/hint/demo/Rectangle.kt:25]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Mkohm/detekt-hint/blob/7e931974fdcda7e78d88743129a7c1a58086cadc/src/main/kotlin/io/github/mkohm/detekt/hint/demo/Rectangle.kt#L25"></span>The file Rectangle.kt is not ending with a new line.</td>
    </tr>
  </tbody>
</table>

File: src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+        property: KtNamedDeclaration,
+        publicMethod: KtNamedFunction,
+        foundCallees: List<KtNamedFunction>
+    ): ArrayList<KtNamedFunction> {
+
+        // If we have any call expressions inside the parent function with the same name as the parent function - There are some possibilities;
+        // 1. calling another function with the same name, but with different signature
+        // 2. calling the supermethod
+        // 3. calles itself - we have recursion.
+        // Todo: create good test cases that will cover these different edge-cases
```
Comment (Detekt-hintBot) [inline src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Mkohm/detekt-hint/blob/7e931974fdcda7e78d88743129a7c1a58086cadc/src/main/kotlin/io/github/mkohm/detekt/hint/rules/LackOfCohesionOfMethodsRule.kt#L151"></span>This comment contains text that has been defined as forbidden in detekt.</td>
    </tr>
  </tbody>
</table>
